### ISSUE TITLE
Hi, the Forecast API only keeps the last couple of months. Afterwards data is archived, but the earliest is July 2022.

### ISSUE THREAD

**Issue by juanjuanqi** [open-meteo/open-meteo#694]
Hi, the Forecast API only keeps the last couple of months. Afterwards data is archived, but the earliest is July 2022.

You can use the [historical weather API](https://open-meteo.com/en/docs/historical-weather-api) with data from 1940 onwards. Unfortunately, the resolution is only 25 km for atmospheric variables like clouds, radiation or wind and 10 km for surface variables like temperature, humidity and soil.

In the next weeks, an additional dataset will follow at 9 km resolution for all weather variables with data from 2017 onwards.

_Originally posted by @patrick-zippenfenig in https://github.com/open-meteo/open-meteo/discussions/353#discussioncomment-6097655_

**Comment by juanjuanqi** [comment]
Hello, I want to download the ECMWF forecast data in 2023 to verify its forecast effect of precipitation. When I select forecast length as 10 days and past days as 0 in the forecast API, I can obtain the forecast data for the next 10 days of ECMWF, and the data is exactly what I need. But when I select forecast length of 10 days and past days as 3 months, I will get data with a length of 102 days. I can understand that the last 10 days are the forecast data of ECMWF for the future. What are the first 90 days, is it forecast data or similar actual weather data?
In short, I would like to obtain ECMWF daily forecast data for the next 10 days in 2023. Can I obtain it from the forcast API.
Thank you for your reply!

**Comment by patrick-zippenfenig** [comment]
Hi, If I understand you correctly, the work-in-progress [Previous Model Runs API](https://open-meteo.com/en/docs/previous-runs-api) could help you. This API continuously stores weather forecasts from a couple of days prior any given date. Is this what you are looking for?
